app/helpers/document_helper.rb

```diff
@@ -78,18 +78,17 @@ def attachment_reference(attachment)
   end
 
   def attachment_thumbnail(attachment)
-    image_url = if attachment.pdf?
-                  attachment.file.thumbnail.url
-                elsif attachment.html?
-                  image_url("pub-cover-html.png", host: Whitehall.public_root)
-                elsif %w[doc docx odt].include? attachment.file_extension
-                  image_url("pub-cover-doc.png", host: Whitehall.public_root)
-                elsif %w[xls xlsx ods csv].include? attachment.file_extension.downcase
-                  image_url("pub-cover-spreadsheet.png", host: Whitehall.public_root)
-                else
-                  image_url("pub-cover.png", host: Whitehall.public_root)
-                end
-    image_tag(image_url, alt: "")
+    if attachment.pdf?
+      image_tag(attachment.file.thumbnail.url, alt: "")
+    elsif attachment.html?
+      render "govuk_publishing_components/components/attachment/thumbnail_document"
+    elsif %w[doc docx odt].include? attachment.file_extension
+      render "govuk_publishing_components/components/attachment/thumbnail_document"
+    elsif %w[xls xlsx ods csv].include? attachment.file_extension.downcase
+      render "govuk_publishing_components/components/attachment/thumbnail_spreadsheet"
+    else
+      render "govuk_publishing_components/components/attachment/thumbnail_generic"
+    end
```

Comment:
I wonder if it's a bit risky to be using these pulled in from publishing_components - they feel a bit like a private API - and should move copies of these SVGs into Whitehall

Reply:
I sort of wondered about this. Ideally I'd use the entire component, but that would mean going one level up and replacing a lot more code which I'm not entirely sure I'm comfortable with. (In particular I think it would mean losing PDF previews, which since they're currently working perfectly feels counterproductive.)

Reply:
Yup, switching to that component seems definitely out of scope. One day!